server/util.go

```diff
@@ -247,10 +247,20 @@ func interceptConfigs(rootViper *viper.Viper, customAppTemplate string, customCo
 			return nil, fmt.Errorf("error in config file: %w", err)
 		}
 
-		conf.RPC.PprofListenAddress = "localhost:6060"
-		conf.P2P.RecvRate = 5120000
-		conf.P2P.SendRate = 5120000
-		conf.Consensus.TimeoutCommit = 5 * time.Second
+		defaultCometCfg := cmtcfg.DefaultConfig()
+		if conf.Consensus.TimeoutCommit == defaultCometCfg.Consensus.TimeoutCommit {
+			conf.Consensus.TimeoutCommit = 5 * time.Second
+		}
```

Comment:
I don't fully understand this. This is what it's doing:
```
if conf.Consensus.TimeoutCommit == 1s {
    conf.Consensus.TimeoutCommit = 5s
}
```

This other seems unnecessary because that's the default value for `RecvRate`, so it's ineffective:

```
if conf.P2P.RecvRate == defaultCometCfg.P2P.RecvRate {
			conf.P2P.RecvRate = 5120000
}
```

Is that expected?

Reply:
Well this is what it was always doing, so I kept the same behavior.

The issue is when you purposely change the timeout config in intercept pre handler, then it was getting overwritten here.
Making the override useless.
So this just checks if you didn't change the config before, so the SDK stays opinionated if the app dev hasn't changed that.

Reply:
Removed the unnecessary override and added a comment.